src/services/mongoose/schema.types.d.ts

```diff
@@ -9,8 +9,8 @@ type IPType = {
   lastConnection: Date
 }
 
-type TwoFA = {
-  secret: string | undefined
+type TwoFAForUser = {
+  secret?: TwoFASecret
```

Comment:
should be `secret: TwoFASecret | null` (maybe should be confirmed in code review)

Reply:
Looking through the codebase I think this `twoFA.secret` point can be a separate cleanup issue because there's some inconsistent logic (sometimes `undefined`, sometimes strict check for `null`) in different places.